### PR TITLE
Anton Kravchenko HW1

### PR DESCRIPTION
**Pull request by anton-kravchenko**

### CODE REVIEW

File: src/anton.kravchenko/HW1/tasks.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  );
+
+/* I've spent roughly 2 hours to add types for the following func and while
+all those "types" are perfectly valid (at least I believe so),
+this makes me think, that the author of 
+https://medium.com/javascript-scene/the-typescript-tax-132ff4cb175b (highly recommend to read)
+is actually right about some "aspects" of strong typing "at scale". */
+
+/* tslint:disable-next-line variable-name */
+const multiplyNumeric___FancyAndTrulyFunctionalVersion = R.pipe<
```
Comment (LimarenkoDenis) [inline src/anton.kravchenko/HW1/tasks.ts]: u can use 
`
IDict<number | any>
`
and then anyway u have number type check

Comment (anton-kravchenko) [reply src/anton.kravchenko/HW1/tasks.ts]: это не спортивно :)

File: package.json [hunk 2]
```diff
@@ -13,20 +13,31 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "rxjs": "^6.3.2"
+    "@types/lodash": "^4.14.122",
```
Comment (LimarenkoDenis) [inline package.json]: recommended to install to devDependencies that u are going to use for development purpose (typescript, @types, tools)

File: package.json [hunk 3]
```diff
@@ -13,20 +13,31 @@
...
+    "@types/ramda": "^0.25.51",
+    "husky": "^1.3.1",
+    "lodash": "^4.17.11",
+    "npm-run-all": "^4.1.5",
+    "prettier": "^1.16.4",
+    "pretty-quick": "^1.10.0",
+    "ramda": "^0.26.1",
+    "rimraf": "^2.6.3",
+    "rxjs": "^6.3.2",
+    "ts-lint": "^4.5.1",
```
Comment (LimarenkoDenis) [inline package.json]: ts-lint?
do u want to install packages for all project only for your local directory?

Comment (anton-kravchenko) [reply package.json]: It was already there and it was just automatically moved from line 28 
https://github.com/LimarenkoDenis/GL-ts-lesson/pull/20/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2L28

Comment (anton-kravchenko) [reply package.json]: Fixed.

File: package.json (hunk 2 above)
Comment (anton-kravchenko) [reply package.json]: Fixed.

File: src/anton.kravchenko/HW1/tasks.ts (hunk 1 above)
Comment (LimarenkoDenis) [reply src/anton.kravchenko/HW1/tasks.ts]: ok, we will discuss it)

**Comment by LimarenkoDenis** [comment]
@anton-kravchenko appreciate your PR and question

**Comment by LimarenkoDenis** [comment]
@anton-kravchenko so, let's delete root dependencies, except upgraded libs, and it'll merge it, 
and we will discuss your notes in the next lesson
are u agree?

**Comment by anton-kravchenko** [comment]
I reverted all of the changes in root package.json except following dependencies: ramda, lodash and types for them. Without these deps my changes won't compile.

**Pull request merged**